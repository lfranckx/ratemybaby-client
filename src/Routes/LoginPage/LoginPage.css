#login-section {
    display: inline-block;
    background-color: white;
    border-radius: 10px;
    padding: 20px;
    box-shadow: 1.5px 3px 5px grey;
    width: 400px;
    color: rgb(41, 39, 39);
    margin-top: 10px;
}

#login-header {
    font-size: 2.5rem;
    opacity: 0.9;    
}

#login-section label {
    color: rgb(41, 39, 39);
}

#login-button {
    margin-top: 50px;
    background-color: rgb(249, 103, 103);
    background-image: linear-gradient(to right, rgb(247, 87, 87) , rgb(249, 147, 103));    
    display: inline-block;
    margin-top: 10px;
    padding: 10px 30px;
    border-radius: 50px;
    border: none;
    color: white;
    cursor: pointer;
    box-shadow: 1px 1px 1px grey;
}

#login-signup-button {
    position: relative;
    top: 100px;
}

.link-container {
    margin-top: 63px;
}

/* MEDIA QUERIES */
@media (max-width: 700px) {
    #login-section {
        margin-top: 0px;
        width: 300px;
    }

    #login-header {
        font-size: 1.5rem;
    }

    #login-form label {
        font-size: 1rem;
    }

    #login-form input {
        font-size: 10px;
    }

    #login-button {
        font-size: 15px;
        padding: 6.5px 35px;
    }

    .link-container {
        margin-top: 20px;
    }
}

.demo {
    text-align: left;
}

@media (max-width: 400px) {
    #login-section {
        margin-top: 0px;
        width: 230px;
    }
}