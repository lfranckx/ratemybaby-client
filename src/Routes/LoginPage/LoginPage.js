import React, { Component } from 'react'
import LoginForm from '../../Components/LoginForm/LoginForm'

export default class LoginPage extends Component {
    static defaultProps = {
      location: {},
      history: {
        push: () => {},
      },
    }
  
    handleLoginSuccess = () => {
      const { location, history } = this.props
      const destination = (location.state || {}).from || '/'
      history.push(destination)
    }
  
    render() {
      return (
        <section>
          <h3>Login</h3>
          <LoginForm onLoginSuccess={this.handleLoginSuccess}/>
        </section>
      )
    }
  }