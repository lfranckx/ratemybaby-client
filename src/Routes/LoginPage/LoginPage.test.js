import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from './LoginPage';

describe(`LoginPage component`, () => {

    it(`renders log in page by default`, () => {
        const wrapper = shallow(<LoginPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})