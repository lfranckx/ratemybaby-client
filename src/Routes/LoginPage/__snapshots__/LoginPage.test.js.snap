// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders log in page by default 1`] = `
<Fragment>
  <section
    id="login-section"
  >
    <h2
      id="login-header"
    >
      Login
    </h2>
    <LoginForm
      onLoginSuccess={[Function]}
    />
  </section>
  <div
    className="link-container"
  >
    <Link
      className="register"
      to="/register"
    >
      Sign Up
    </Link>
  </div>
</Fragment>
`;
